sphere-models/src/it/java/io/sphere/sdk/products/attributes/LocalizedToStringProductAttributeConverterTest.java

```diff
@@ -0,0 +1,222 @@
+package io.sphere.sdk.products.attributes;
+
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductsScenario1Fixtures;
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.test.IntegrationTest;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Locale;
+
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static io.sphere.sdk.products.ProductsScenario1Fixtures.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LocalizedToStringProductAttributeConverterTest extends IntegrationTest {
+
+    private static ProductsScenario1Fixtures.Data data;
+    private static Product product;
+    private static ProductType productType;
+    private static ProductAttributeConverter<String> mapper;
+
+    @BeforeClass
+    public static void setupScenario() {
+        data = ProductsScenario1Fixtures.createScenario(client());
+        product = data.getProduct1();
+        productType = data.getProductType();
+        mapper = new LocalizedToStringProductAttributeConverter(singletonList(productType), singletonList(Locale.GERMANY)) {
+            @Override
+            protected Collection<String> longAttributes() {
+                return singletonList(ATTR_NAME_NUMBER);
+            }
+
+            @Override
+            protected Collection<String> longSetAttributes() {
+                return singleton(ATTR_NAME_NUMBER_SET);
+            }
+        };
+    }
+
+    @AfterClass
+    public static void cleanupScenario() {
+        ProductsScenario1Fixtures.destroy(client(), data);
+    }
+
+    @Test
+    public void convertStringSet() {
+        assertThat(converting(ATTR_NAME_TEXT_SET)).isEqualTo("bar, foo");
+    }
+
+    @Ignore
+    @Test
+    public void convertProductTypeReferenceSet() {
+    }
+
+    @Ignore
+    @Test
+    public void convertProductTypeReference() {
+    }
+
+    @Test
+    public void convertProductReferenceSet() {
+        assertThat(converting(ATTR_NAME_REF)).isNull();
+    }
+
+    @Test
+    public void convertProductReference() {
+        assertThat(converting(ATTR_NAME_REF_SET)).isNull();
```

Comment:
I would print the reference ID. I know it's not very usable, but with null we might have two bad scenarios: 1) NPE or 2) nothing shown. I would rather have the ID so at least people can obtain results (though they probably need to be adjusted).